code/game/objects/items/storage/holsters.dm

```diff
@@ -0,0 +1,255 @@
+		icon_state = base_icon + "_full"
+		item_state = icon_state
+	else
+		playsound(src,drawSound, 15, 1)
+		icon_state = base_icon
+		item_state = icon_state
+	//actually updates the icon
+	update_icon()
+
+/obj/item/storage/holster/update_icon()
+	. = ..()
+	var/mob/user = loc
+	if (!istype(user))
+		return
+	user.update_inv_back()
+	user.update_inv_belt()
+	user.update_inv_s_store()
+
+//backpack type holster items
+/obj/item/storage/holster/backholster
+	name = "backpack holster"
+	desc = "You wear this on your back and put items into it. Usually one special item too."
+	sprite_sheets = list("Combat Robot" = 'icons/mob/species/robot/backpack.dmi')
+	max_w_class = 3 //normal items
+	max_storage_space = 24
+	access_delay = 1.5 SECONDS ///0 out for satchel types
+
+//only applies on storage of all items, not withdrawal
+/obj/item/storage/holster/backholster/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if (!use_sound)
+		return
+	playsound(loc, use_sound, 15, 1, 6)
+
+/obj/item/storage/holster/backholster/equipped(mob/user, slot)
+	if(slot == SLOT_BACK)
+		mouse_opacity = 2 //so it's easier to click when properly equipped.
+		if(use_sound)
+			playsound(loc, use_sound, 15, 1, 6)
+	return ..()
+
+
+///RR bag
+/obj/item/storage/holster/backholster/rpg
+	name = "\improper TGMC rocket bag"
+	desc = "This backpack can hold 5 67mm shells or 80mm rockets, in addition to a recoiless launcher."
+	icon_state = "marine_rocket"
+	item_state = "marine_rocket"
+	base_icon = "marine_rocket"
+	w_class = WEIGHT_CLASS_HUGE
+	storage_slots = 5
+	max_w_class = 4
+	access_delay = 0.5 SECONDS
+	holsterable_allowed = list(/obj/item/weapon/gun/launcher/rocket/recoillessrifle,)
+	bypass_w_limit = list(/obj/item/weapon/gun/launcher/rocket/recoillessrifle,)
+	///only one RR per bag
+	storage_type_limits = list(/obj/item/weapon/gun/launcher/rocket/recoillessrifle = 1,)
+	can_hold = list(
+		/obj/item/ammo_magazine/rocket,
+		/obj/item/weapon/gun/launcher/rocket/recoillessrifle,
+	)
+	sprite_sheets = list("Combat Robot" = 'icons/mob/species/robot/backpack.dmi') //robots have their own snowflake back sprites
+
+//one slot holsters
+
+///swords
+/obj/item/storage/holster/blade
+	///used only for storage path purposes
+	name = "\improper default holster"
+	desc = "You shouldn't see this."
+
+/obj/item/storage/holster/blade/machete
+	name = "\improper H5 pattern M2132 machete scabbard"
+	desc = "A large leather scabbard used to carry a M2132 machete. It can be strapped to the back, waist or armor."
+	icon_state = "machete_holster"
+	base_icon = "machete_holster"
+	flags_equip_slot = ITEM_SLOT_BELT|ITEM_SLOT_BACK
+	holsterable_allowed = list(
+		/obj/item/weapon/claymore/mercsword/machete,
+		/obj/item/weapon/claymore/harvester,
+	)
+	can_hold = list(
+		/obj/item/weapon/claymore/mercsword/machete,
+		/obj/item/weapon/claymore/harvester,
+	)
+
+/obj/item/storage/holster/blade/machete/full/Initialize()
+	. = ..()
+	icon_state = "machete_holster_full"
```

Comment:
Since you are touching, you are fixing :)

This should be set on the object definition, not at runtime